docs: Add AGENTS.md with repository instructions for AI tooling (#1516)
munich-quantum-toolkit/core: AGENTS.md

# MQT Core

- Acts as the high-performance backbone for the [Munich Quantum Toolkit (MQT)](https://mqt.readthedocs.io/).
- Provides core data structures and algorithms (IR, DD, ZX, QASM 3.0, MLIR) for quantum design automation.
- Distinguishes between **circuit qubits** (logical/virtual) and **device qubits** (physical) to avoid ambiguity.
- Maintains a C++20 core with an architecturally separate Python extension layer.
- Locates C++ sources in `src/` and `include/`, Python bindings in `python/`, tests in `test/`, and documentation in `docs/`.
- Refer to [index.md](docs/index.md) for the complete documentation entry point.

## Environment & Capabilities

- **Platform**: Operating in a Linux environment with support for x86_64 (amd64) and arm64.

> - **Python Context**: Version 3.10+ managed via **uv** for strict isolation.

Reviewer: ```suggestion
- **Python Context**: Version 3.10+ managed via **uv**.
```
I don't particularly know what the "strict isolation" part means here.

- **C++ Context**: C++20 standard required; **LLVM 21.1+** mandatory for MLIR dialects.
- **Tooling Access**: Full access to `cmake`, `ctest`, `pytest`, `ruff`, and `nox`.
- **Constraint**: Avoid network-dependent tasks during builds; rely on `uv` lockfiles and pre-synced dependencies.

## Repository Mapping

- `include/mqt/core/`: C++ header files (Internal & Public interfaces).

> - `src/mqt/core/`: C++ implementation logic.

Reviewer: this is not correct. the corresponding directory does not even exist.

- `python/mqt/core/`: Python package foundation.
- `python/mqt/core/ir/`: Public Python bindings package for Circuit IR.
- `python/mqt/core/dd.pyi`: Public Python bindings module for Decision Diagrams.

> - `python/mqt/core/na/`: Public Python bindings package for Neutral Atom logic.

Reviewer: This is wildly different in granularity than the rest. The C++ part also does not show any modules for example.

- Use **nanobind** (~2.11.0) for Python bindings; do not use `pybind11`.
- Use **ruff** for linting and formatting, with `ALL` rules enabled by default.
- Use **ty** for static type checking as the mandatory replacement for `mypy`.
- Use **LLVM 21.1+** for MLIR-based compilation dialects.
- Use **clang-format** and **typos** for C++ style and project-wide spell checking.

## Development Guidelines

- Always prioritize C++20 `std::` features over custom implementations.
- Use Google-style docstrings for Python and Doxygen comments for C++.
- Ensure Python code is fully typed and passes `ty` static analysis.
- Follow existing code style by checking neighboring files for patterns.
- Add or update tests for every code change, even if not explicitly requested.
- Review [CHANGELOG.md](docs/CHANGELOG.md) and [UPGRADING.md](docs/UPGRADING.md) before making breaking changes.

## Documentation Reference

- Deep dive into the Internal Representation: [mqt_core_ir.md](docs/mqt_core_ir.md).
- Understand Decision Diagram internals: [dd_package.md](docs/dd_package.md).
- Explore ZX-calculus algorithms: [zx_package.md](docs/zx_package.md).

> - Access the online API reference: [MQT Core Docs](https://mqt.readthedocs.io/projects/core/en/latest/).

Reviewer: ```suggestion
```
Does not feel too useful for day to day use.

# MQT Core

- Acts as the high-performance backbone for the [Munich Quantum Toolkit (MQT)](https://mqt.readthedocs.io/).
- Provides tools and methods (IRs, DD, ZX, OpenQASM, MLIR, QIR, QDMI, NA, Qiskit, etc.) for quantum design automation, including simulation, compilation, verification as well as HPCQC integration.
- Maintains a C++20 core with an architecturally separate Python extension layer.
- Locates C++ sources in `src/` and `include/`, MLIR sources in `mlir/`, Python bindings in `bindings/`, the Python package in `python/mqt/core/`, tests in `test/`, and documentation in `docs/`.

## Environment & Capabilities

- **Platform**: Target Linux, macOS, and Windows for x86_64 (amd64) and arm64.
- **Python Context**: Version 3.10+ managed via **uv**.
- **C++ Context**: C++20 standard required; **LLVM 21.1+** mandatory for MLIR infrastructure.

> - **Tooling Access**: Full access to `cmake`, `ctest`, `uv`, and `nox`.

Reviewer: ```suggestion
- **Tooling Access**: Full access to `uv`. `cmake`, `nox`, and further tools are available through `uv` (via `uvx <tool>`)
```
Generalizing this a bit.

## Repository Mapping

> - `include/mqt/core/`: public and internal C++ header files.

Reviewer: ```suggestion
- `include/mqt-core/`: public C++ header files.
```
the path here was wrong and this contains all the public headers
